### PR TITLE
This can realize listing images and containers using React and JSX. Please review.

### PR DESCRIPTION
**Pull request by QiWang19**
- Moved from react-lite to React 16.0.0. 
- Copied cockpit-components-listing.jsx from DockerUI but changed to adjust to React16
- Please point out if I'm in the wrong way

### CODE REVIEW

**Comment by martinpitt** [comment]
You need to rebase against current master. Also, time to turn CI back on, it should work now :-) I'll do that.

File: Makefile [hunk 1]
```diff
@@ -82,7 +82,7 @@ devel-install: dist/index.js
 
 # when building a distribution tarball, call webpack with a 'production' environment
 dist-gzip: NODE_ENV=production
-dist-gzip: clean all
+dist-gzip: clean $(NODE_MODULES_TEST) all
```
Comment (martinpitt) [inline Makefile]: No, please drop this. This got deliberately [dropped in starter-kit](https://github.com/cockpit-project/starter-kit/commit/e9635620c57b78) as it breaks RPM builds that have no  network. Current cockpit-podman master is rebased against current starter-kit, and `dist-gzip` does exactly what it should now.  So please just rebase your branch against current origin/master.

Also, please don't do commits with such commit messages. They should describe *what* changed in high-level terms, and also *why* the change was made (unless it's obvious, like typo fixes).

File: package.json [hunk 2]
```diff
@@ -26,14 +26,20 @@
     "jed": "^1.1.1",
     "jshint": "~2.9.1",
     "jshint-loader": "~0.8.3",
+    "less": "^3.7.1",
+    "less-loader": "^4.1.0",
```
Comment (martinpitt) [inline package.json]: Such "merge upstream" commits must not appear in PRs, they are essentially impossible to review. Please use `git fetch origin; git rebase -i origin/master` ("rebasing") to put your changes on top of what's already in master, and then adjust *your* commits (not the master ones).

File: src/index.es6 [hunk 3]
```diff
@@ -20,7 +20,7 @@
 
 import React from 'react';
 import Application from './app.jsx';
-// import Overview from './app.jsx';
+
```
Comment (martinpitt) [inline src/index.es6]: This line is not in master, so some other commit of your's in this PR already adds this. These "fixups" should be squashed into the commit that they fix. The goal is a PR is a series of self-contained, correct, and well-described commits, which make sense by themselves, and can be reviewed and understood.

**Comment by martinpitt** [comment]
@QiWang19 : Please see the "show outdated" comments above.

**Comment by martinpitt** [comment]
Sorry, this cannot currently be reviewed. Neither the per-commit view nor the "all changes" view provide reasonable diffs, so this really needs to be rebased and squashed first.

**Comment by QiWang19** [comment]
Closing replaced with https://github.com/cockpit-project/cockpit-podman/pull/3

**Comment by martinpitt** [comment]
Note that you can just force-push to your branch to update an existing PR. There's no need to create a new one (it doesn't hurt much either, though).

**Comment by rhatdan** [comment]
@martinpitt I was helping her clean this up.  The first PR was master on master, so I figured I would show her how to create a branch and submit from there.

**Pull request closed without merging**